### PR TITLE
docs: Add kubernetes azure auth provider section

### PR DESCRIPTION
**Pull request by luchillo17**
Signed-off-by: Carlos Esteban Lopez <lcarlosesteb@vmware.com>

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/features/kubernetes/authentication.md [hunk 1]
```diff
@@ -28,6 +28,28 @@ The providers available as server side are:
 - `localKubectlProxy`
 - `serviceAccount`
 
+### Azure
+
+The Azure server side authentication provider works by authenticating on the server with
+the Azure CLI `az login`, meaning any user that can access the backstage app will have
+the same level of access to the cluster in the Kubernetes plugin, to put an example, even
+guest users would have access to the cluster resources.
```
Comment (jamieklassen) [inline docs/features/kubernetes/authentication.md:36]: This is actually true for any server side auth provider, so maybe call it out in the above section rather than here?

File: docs/features/kubernetes/authentication.md [hunk 2]
```diff
@@ -28,6 +28,28 @@ The providers available as server side are:
 - `localKubectlProxy`
 - `serviceAccount`
 
+### Azure
+
+The Azure server side authentication provider works by authenticating on the server with
+the Azure CLI `az login`, meaning any user that can access the backstage app will have
```
Comment (jamieklassen) [inline docs/features/kubernetes/authentication.md:34]: I think the truth is a bit more complicated than "it uses `az login`" -- I'd love to get input from @goenning who added this auth provider, but in [this comment](https://github.com/backstage/backstage/pull/11299#issuecomment-1119584630) they describe some of the subtleties of that `DefaultAzureCredential` from microsoft. Technically it _can_ use `az login` if the azure CLI is available to the backstage process, but it will try getting creds from [`AZURE_*` env vars](https://docs.microsoft.com/en-us/dotnet/api/azure.identity.environmentcredential?view=azure-dotnet) or [Azure's Managed Identity service](https://docs.microsoft.com/en-us/dotnet/api/azure.identity.managedidentitycredential?view=azure-dotnet) first, according to [these docs](https://docs.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet).

File: docs/features/kubernetes/authentication.md [hunk 3]
```diff
@@ -28,6 +28,28 @@ The providers available as server side are:
...
+the same level of access to the cluster in the Kubernetes plugin, to put an example, even
+guest users would have access to the cluster resources.
+
+```yaml
+kubernetes:
+  clusterLocatorMethods:
+    - type: 'config'
+      clusters:
+        - name: Random cluster name in backstage
```
Comment (jamieklassen) [inline docs/features/kubernetes/authentication.md:43]: Maybe
```suggestion
        - name: My AKS cluster
```
would be a little more helpful?

File: docs/features/kubernetes/authentication.md (hunk 2 above)
Comment (jamieklassen) [reply docs/features/kubernetes/authentication.md:34]: Also maybe worth noting (though it's quite technical and specific) that azure CLI command that gets run by the `AzureCliCredential` is not `az login`, but [`az account get-access-token --output json --resource 6dae42f8-4368-4678-94ff-3960e28e3630`](https://github.com/Azure/azure-sdk-for-js/blob/eba5089bba45358b62f1d3b1b7c33972f224e812/sdk/identity/identity/src/credentials/azureCliCredential.ts#L50-L59), where that GUID is the ID of the "Azure Kubernetes Service AAD Server" used by all AAD-enabled AKS clusters.

**Comment by jamieklassen** [comment]
Can we call out the fact that "Azure cluster" means AKS? Also, something important we learned about using this auth provider with AKS is that, because it works with bearer tokens, it only works if [Azure AD Authentication](https://docs.microsoft.com/en-us/azure/aks/managed-aad) is enabled on the cluster -- can we mention that too?

File: docs/features/kubernetes/authentication.md (hunk 1 above)
Comment (luchillo17) [reply docs/features/kubernetes/authentication.md:36]: I only got to test Azure clusters with the guest login, so I was not sure if it was indeed the case for all server-side providers, I'll move & reword this to the server-side section directly.

File: docs/features/kubernetes/authentication.md (hunk 2 above)
Comment (luchillo17) [reply docs/features/kubernetes/authentication.md]: I've updated the text & steps, should I go in detail about those other ways to provide credentials?

Comment (jamieklassen) [reply docs/features/kubernetes/authentication.md]: The extra details here are very nice, props.

I know you didn't actually explore this in your investigations, but unfortunately the azure CLI use case is probably not the most important one to document -- it would mainly be used when iterating on a Backstage app running locally. The env var and managed identity options are more suitable for production use cases -- I imagine that's what @goenning uses for real.

You probably don't want to document something you haven't actually tested yourself, and I don't think it's worth investing your time in testing those other cases at this point, so I'm happy resolving this conversation for now. It's a big improvement!

File: docs/features/kubernetes/authentication.md [hunk 4]
```diff
@@ -18,7 +18,9 @@ The providers currently available are divided into server side and client side.
 ## Server Side Providers
 
 These providers authenticate your _application_ with the cluster, meaning anyone that is
-logged in into your backstage app will be granted the same access to Kubernetes objects.
+logged in into your backstage app will be granted the same access to Kubernetes objects,
+to put an example,even guest users would have access to the cluster resources, so be
+careful of using server side providers with guest access enabled.
```
Comment (Rugvip) [inline docs/features/kubernetes/authentication.md:23]:
```suggestion
logged in into your backstage app will be granted the same access to Kubernetes objects, including guest users.
```

File: docs/features/kubernetes/authentication.md [hunk 5]
```diff
@@ -28,6 +30,33 @@ The providers available as server side are:
 - `localKubectlProxy`
 - `serviceAccount`
 
+### Azure
+
+The Azure server side authentication provider works by authenticating on the server with
+the Azure CLI, please note that [Azure AD Authentication][1] is a requirement and has to
+be enabled in your AKS cluster, then follow these steps:
+
+- [Install the Azure CLI][2]
```
Comment (Rugvip) [inline docs/features/kubernetes/authentication.md:39]: I think it could be useful to clarify where the CLI should be installed

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 

Couple of small nits

File: docs/features/kubernetes/authentication.md [hunk 6]
```diff
@@ -28,6 +28,33 @@ The providers available as server side are:
 - `localKubectlProxy`
 - `serviceAccount`
 
+### Azure
+
+The Azure server side authentication provider works by authenticating on the server with
+the Azure CLI, please note that [Azure AD Authentication][1] is a requirement and has to
+be enabled in your AKS cluster, then follow these steps:
+
+- [Install the Azure CLI][2] in the device where the backstage application will run.
```
Comment (Rugvip) [inline docs/features/kubernetes/authentication.md:37]:
```suggestion
- [Install the Azure CLI][2] in the environment where the backstage application will run.
```

Sorry device was a bit too awkward for me 😅

**Review by Rugvip — APPROVED** [review]
Thank you! 👍 

Gonna apply my own suggestion and then :shipit: 🎉
